snips-nlu-lib/src/injection/injection.rs

```diff
@@ -204,32 +297,111 @@ fn get_nlu_engine_metadata<P: AsRef<Path>>(engine_dir: P) -> Result<NLUEngineInf
             msg: format!("invalid nlu engine model file in {:?}", engine_dataset_metadata_path)
         })?;
 
+    let language = NluUtilsLanguage::from_str(&*nlu_engine_model.dataset_metadata.language_code)
+        .map_err(|_| NluInjectionErrorKind::InternalInjectionError {
+            msg: "invalid nlu engine language".to_string()
+        })?;
+
     let custom_entities = HashSet::from_iter(
         nlu_engine_model.dataset_metadata.entities.keys().map(|k| k.clone()));
 
     let builtin_entity_parser_dir = engine_dir.as_ref().join(nlu_engine_model.builtin_entity_parser);
     let custom_entity_parser_dir = engine_dir.as_ref().join(nlu_engine_model.custom_entity_parser);
 
-    Ok(NLUEngineInfo {
+    Ok(NluEngineInfo {
+        language,
         builtin_entity_parser_dir,
         custom_entity_parser_dir,
         custom_entities,
     })
 }
 
+fn normalize_entity_value(
+    entity_values: Vec<InjectedValue>
+) -> Vec<GazetteerEntityValue> {
+    entity_values
+        .into_iter()
+        .map(|value| GazetteerEntityValue {
+            raw_value: normalize(&*value),
+            resolved_value: value.to_string(),
+        })
+        .collect()
+}
+
+fn stem_entity_value(
+    entity_values: Vec<GazetteerEntityValue>,
+    engine_info: &NluEngineInfo,
+    custom_entity_parser_info: &CustomGazetteerParserInfo,
+    maybe_stemmer: &Option<&Arc<Stemmer>>,
+) -> Result<Vec<GazetteerEntityValue>, NluInjectionError> {
+    let stemmed_values = match custom_entity_parser_info.parser_usage {
+        CustomEntityParserUsage::WithoutStems => Ok(vec![]),
```

Comment:
I think you can just return `vec![]` here, and remove the `Ok(...)` in the other arm as well.

Reply:
The other arms returns a result because the `stemmer` can be missing, I need to return a `Result`